graphql-dgs-spring-webflux-autoconfigure/src/main/kotlin/com/netflux/graphql/dgs/webflux/handlers/DgsReactiveWebsocketHandler.kt

```diff
@@ -0,0 +1,166 @@
+                                            webSocketSession
+                                        )
+                                    }.doOnSubscribe {
+                                        if (operationMessage.id != null) {
+                                            subscriptions[webSocketSession.id] = mutableMapOf(operationMessage.id to it)
+                                        }
+                                    }.doOnComplete {
+                                        webSocketSession.send(
+                                            Flux.just(
+                                                toWebsocketMessage(
+                                                    OperationMessage(GQL_COMPLETE, null, operationMessage.id),
+                                                    webSocketSession
+                                                )
+                                            )
+                                        ).subscribe()
+
+                                        subscriptions[webSocketSession.id]?.remove(operationMessage.id)
+                                        logger.debug("Completing subscription ${operationMessage.id} for subscription ${operationMessage.id} for connection ${webSocketSession.id}")
+                                    }.doOnError {
+                                        webSocketSession.send(
+                                            Flux.just(
+                                                toWebsocketMessage(
+                                                    OperationMessage(GQL_ERROR, DataPayload(null, listOf(it.message!!)), operationMessage.id),
+                                                    webSocketSession
+                                                )
+                                            )
+                                        ).subscribe()
+
+                                        subscriptions[webSocketSession.id]?.remove(operationMessage.id)
+                                        logger.debug("Subscription publisher error for input $queryPayload for subscription ${operationMessage.id} for connection ${webSocketSession.id}", it)
+                                    }
+                                }
+                        }
+
+                        GQL_STOP -> {
+                            subscriptions[webSocketSession.id]?.get(operationMessage.id)?.cancel()
+                            subscriptions[webSocketSession.id]?.remove(operationMessage.id)
+                            logger.debug("Client stopped subscription ${operationMessage.id} for connection ${webSocketSession.id}")
+                            Flux.empty()
+                        }
+
+                        GQL_CONNECTION_TERMINATE -> {
+                            subscriptions[webSocketSession.id]?.values?.forEach { it.cancel() }
+                            webSocketSession.close()
```

Comment:
Remove the session from subscriptions as well? Like in line 122.

Reply:
done